helpers/api_helper/json_api_response.go

```diff
@@ -18,13 +20,25 @@ func (e *ErrorResponseObject) NewErrorResponse() gin.H {
 }
 
 type DataResponseObject struct {
-	ID         string      `json:"id"`
-	Type       string      `json:"type"`
-	Attributes interface{} `json:"attributes"`
+	ID            string      `json:"id"`
+	Type          string      `json:"type"`
+	Attributes    interface{} `json:"attributes,omitempty"`
+	Relationships interface{} `json:"relationships,omitempty"`
 }
 
-func (d *DataResponseObject) ConstructDataResponse() gin.H {
-	return gin.H{
-		"data": d,
+type DataResponse struct {
+	Data DataResponseObject `json:"data"`
+}
+
+type DataResponseArray struct {
```

Comment:
Instead of `Array`, which is not a concept in Go by the way, how about using `DataResponseList` or `DataResponseCollection`?

Reply:
Good point. Updated in 0f3ebc3a554316431289d265df778edcb2f0ca42